src/components/NcModal/NcModal.vue

```diff
@@ -3,885 +3,549 @@
+
+	/**
+	 * Allow to pause an ongoing slideshow
+	 */
+	slideshowPaused?: boolean
+
+	/**
+	 * Disable swipe between slides
+	 */
+	disableSwipe?: boolean
+
+	/**
+	 * Enable spread navigation
+	 */
+	spreadNavigation?: boolean
+
+	/**
+	 * Defines the modal size.
+	 * All sizes except 'small' change automatically to full-screen on mobile.
+	 */
+	size?: 'small' | 'normal' | 'large' | 'full'
+
+	/**
+	 * Do not show the close button for the dialog.
+	 */
+	noClose?: boolean
+
+	/**
+	 * Close the modal if the user clicked outside the modal
+	 * Only relevant if `noClose` is not set.
+	 */
+	closeOnClickOutside?: boolean
+
+	/**
+	 * Makes the modal backdrop opaque if true
+	 * Will be overwritten if some buttons are shown outside
+	 */
+	dark?: boolean
+
+	/**
+	 * Set light backdrop. Makes the modal header appear light.
+	 */
+	lightBackdrop?: boolean
+
+	/**
+	 * Selector for the modal container, pass `null` to prevent automatic container mounting
+	 */
+	container?: string | null
+
+	/**
+	 * Pass in `true` if you want the modal 'close' button to be displayed
+	 * outside the modal boundaries, in the top right corner of the window.
+	 *
+	 * @since 8.25.0
+	 */
+	closeButtonOutside?: boolean
+
+	/**
+	 * Additional elements to add to the focus trap
+	 */
+	additionalTrapElements?: (string | HTMLElement)[]
+
+	/**
+	 * Display x items inline
+	 *
+	 * @see NcActions component usage
+	 */
+	inlineActions?: number
+
+	/**
+	 * The current open property of the modal
+	 */
+	show?: boolean | undefined
+
+	/**
+	 * Id of the element that labels the dialog (the name)
+	 * Not needed if the `name` prop is set, but if no name is set you need to provide the ID of an element to label the dialog for accessibility.
+	 */
+	labelId?: string
+
+	/**
+	 * Set element to return focus to after focus trap deactivation
+	 */
+	setReturnFocus?: FocusTargetValueOrFalse
+}>(), {
+	additionalTrapElements: () => [],
+	container: 'body',
+	inlineActions: 0,
+	labelId: '',
+	slideshowDelay: 5000,
+	size: 'normal',
+	name: '',
+	show: undefined,
+	setReturnFocus: undefined,
+})
+
+const emit = defineEmits<{
+	/**
+	 * Trigger showing the next slide.
+	 *
+	 * @param payload - The event that triggered showing the next slide
+	 */
+	next: [payload?: Event]
+
+	/**
+	 * Trigger showing the previous slide.
+	 *
+	 * @param payload - The event that triggered showing the previous slide
+	 */
+	previous: [payload?: Event]
+
+	/**
+	 * Emitted when the closing animation is finished
+	 *
+	 * @param payload - The event that triggered the close
+	 */
+	close: [payload?: Event]
+
+	/**
+	 * @param payload - The new show-state
+	 */
+	'update:show': [payload: boolean]
+}>()
+
+defineSlots<{
+	/**
+	 * Actions to show (one or more NcAction* components)
+	 */
+	actions?: Slot
+
+	/**
+	 * The modal content to show.
+	 */
+	default?: Slot
+}>()
+
+const modalId = createElementId()
+const maskElement = useTemplateRef<HTMLDivElement>('mask')
+
+const internalShow = ref(true)
```

Comment:
It seems to me we can simplify the component by just using `defineModel`. Then we don't need the second state, and the rest around the value is simpler. Is there anything blocking it I'm missing?